File: benchmarkDS/ourMethod/GEMINI_gemini-2.5-pro/Lse13.py
import sys

# A predefined sequence of IDs.
id_sequence = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]

def get_value_at_index(index):
    """
    Returns the value at the specified index in the global id_sequence.

    Args:
        index (int): The index of the element to retrieve.

    Returns:
        int: The value at the specified index.

    Raises:
        TypeError: If the index is not an integer.
        IndexError: If the index is out of the valid range.
    """
    # Rule #4: Ensure all input is validated.
    if not isinstance(index, int):
        raise TypeError("Error: Index must be an integer.")
    
    # Rule #1: Ensure that all buffer operations strictly adhere to the buffer's boundaries.
    if 0 <= index < len(id_sequence):
        return id_sequence[index]
    else:
        raise IndexError(f"Error: Index {index} is out of bounds.")

def main():
    """
    Main function to run test cases.
    """
    # 5 test cases, including valid, invalid, and wrong type.
    test_indices = [0, 5, 9, -1, 10, 'a']

    print("Running test cases...")
    for i, test_index in enumerate(test_indices):
        print(f"\n--- Test Case {i + 1} ---")
        print(f"Attempting to access index: {repr(test_index)}")
        try:
            # Rule #7: Ensure that all exceptions are caught and handled appropriately.
            value = get_value_at_index(test_index)
            print(f"Value at index {test_index}: {value}")
        except (IndexError, TypeError) as e:
            # Writing to stderr for errors is a common practice.
            print(f"{e}", file=sys.stderr)

if __name__ == "__main__":
    main()

File: benchmarkDS/ourMethod/GEMINI_gemini-2.5-pro/Lse13.java
import java.util.InputMismatchException;

public class Lse13 {

    private static final int[] id_sequence = {10, 20, 30, 40, 50, 60, 70, 80, 90, 100};

    /**
     * Returns the value at the specified index in the id_sequence array.
     *
     * @param index The index of the element to retrieve.
     * @return The value at the specified index.
     * @throws ArrayIndexOutOfBoundsException if the index is out of range
     *         (index < 0 || index >= id_sequence.length).
     */
    public static int getValueAtIndex(int index) {
        // Rule #1, #4: Ensure that buffer operations strictly adhere to the buffer's boundaries.
        if (index < 0 || index >= id_sequence.length) {
            throw new ArrayIndexOutOfBoundsException("Error: Index " + index + " is out of bounds.");
        }
        return id_sequence[index];
    }

    /**
     * Main method to run test cases.
     *
     * @param args Command line arguments (not used).
     */
    public static void main(String[] args) {
        // 5 test cases
        int[] testIndices = {0, 5, 9, -1, 10};

        System.out.println("Running 5 test cases...");
        for (int i = 0; i < testIndices.length; i++) {
            int testIndex = testIndices[i];
            System.out.println("\n--- Test Case " + (i + 1) + " ---");
            System.out.println("Attempting to access index: " + testIndex);
            try {
                // Rule #7: Ensure that all exceptions are caught and handled appropriately.
                int value = getValueAtIndex(testIndex);
                System.out.println("Value at index " + testIndex + ": " + value);
            } catch (ArrayIndexOutOfBoundsException e) {
                System.err.println(e.getMessage());
            }
        }
    }
}